src/redux/features/books/booksSlice.js

```diff
@@ -1,42 +1,104 @@
 /* eslint-disable no-param-reassign */
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-// Initial state
-const initialState = [
-  {
-    id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
+const baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/xOI7HhKVUsDCTkv7qbXd/books';
+
+// =============== Asynchronous =============
+
+export const getBookFromAPI = createAsyncThunk(
```

Comment:
- [ ] Hi @kensteph, it seems the getBookFromAPI function was not dispatched in your BookList component. As requested in the requirements of this project lets us kindly avoid unwanted component renderings by using the useEffect Hook to dispatch the getBookFromAPI function. **Hint**

```js
import { fetchBooks } from "................."

function BookList() {
  const dispatch = useDispatch()


  useEffect(() => {
    dispatch(getBookFromAPI())
  }, [])

  return <div>{/* BookList JSX including books data */}</div>
}


```